src/fetch/cow_citizens.py

```diff
@@ -0,0 +1,84 @@
+"""
+Stores the result of querying entire GNO token holders on both networks into a single file
+`data/{network}-lp-holders.csv`
+"""
+from __future__ import annotations
+
+import math
+
+from src.dune_analytics import DuneAnalytics
+
+
+def fetch_cow_citizens(
+        dune: DuneAnalytics,
+        investment_threshold: float = 0.8
+) -> (list[dict], list[dict]):
+    """
+    Queries for investors and generates a
+    :param dune: open connection to dune analytics
+    :param investment_threshold: percentage of exercising dominant investment.
+    :return: collection of metadata data related minting of CoW Citizen NFT on `network`
+    """
+    dune_citizens = []
+    for network in ['mainnet', 'gchain']:
+        citizens.extend(dune.fetch(
+            query_filepath=f"./queries/{network}_citizens.sql",
+            network=network,
+            name="CoW Citizens",
+            parameters=[
+                {
+                    "key": "InvestmentThreshold",
+                    "type": "number",
+                    "value": str(investment_threshold)
+                }
+            ]
+        ))
+
+    dune_citizens.sort(key=lambda t: t['claim_index'])
+    # The following assertion implies that there was no collision of
+    # claim_index after merging the two result files
+    assert len(dune_citizens) == len(set(t['claim_index'] for t in dune_citizens))
```

Comment:
We could also insert a chain parameter into the claim_index, in order to prevent this, if it ever becomes an issue. But if the assertion passes, then there is no need and all good.

Reply:
Cool idea, but the assertion passes so there is no need to do it here!